Resources/config/orm.xml

```diff
@@ -102,6 +102,17 @@
             <factory class="%doctrine.orm.entity_manager.class%" method="create" />
         </service>
 
+        <service id="doctrine.orm.container_repository_factory" class="Doctrine\Bundle\DoctrineBundle\Repository\ContainerRepositoryFactory" public="false">
+            <argument type="service">
+                <service class="Symfony\Component\DependencyInjection\ServiceLocator">
```

Comment:
won't work before Symfony DI 3.3. Min version is 2.7

Reply:
And may add an id on this service to avoid:
```php
        $locatorDef = $container->findDefinition($container->getDefinition('doctrine.orm.container_repository_factory')->getArgument(0));
```
It will be easier to understand